nexus/tests/integration_tests/device_auth.rs

```diff
@@ -633,6 +633,262 @@ async fn test_device_token_request_ttl(cptestctx: &ControlPlaneTestContext) {
         .expect("token should be expired");
 }
 
+#[nexus_test]
+async fn test_device_token_cannot_extend_expiration(
+    cptestctx: &ControlPlaneTestContext,
+) {
+    let testctx = &cptestctx.external_client;
+
+    // get silo belonging to privileged user to make sure local user for session
+    // testing is created in the same silo
+    let me = object_get::<views::CurrentUser>(testctx, "/v1/me").await;
+    let silo_name = me.silo_name.as_str();
+
+    // Set silo max TTL to 15 seconds
+    let settings = params::SiloAuthSettingsUpdate {
+        device_token_max_ttl_seconds: NonZeroU32::new(15).into(),
+    };
+    let _: views::SiloAuthSettings =
+        object_put(testctx, "/v1/auth-settings", &settings).await;
+
+    // First, test that session auth does NOT clamp token TTL
+    // Create a local user and get a session token
+    let silo_url = format!("/v1/system/silos/{}", silo_name);
```

Comment:
nitpick; considered pedantic:
```suggestion
    let silo_url = format!("/v1/system/silos/{silo_name}");
```